source/slang/slang-emit-c-like.cpp

```diff
@@ -3704,6 +3714,234 @@ void CLikeSourceEmitter::emitFuncDecorationsImpl(IRFunc* func)
     }
 }
 
+bool CLikeSourceEmitter::tryGetIntInfo(IRType* elementType, bool &isSigned, int &bitWidth)
+{
+    Slang::IROp type = elementType->getOp();
+    if (!(type >= kIROp_Int8Type && type <= kIROp_UInt64Type)) return false;
+    isSigned = (type >= kIROp_Int8Type && type <= kIROp_Int64Type);
+
+    Slang::IROp stype = (isSigned) ? type : Slang::IROp(type - 4);
+    bitWidth = 8 << (stype - kIROp_Int8Type);
+    return true;
+}
+
+void CLikeSourceEmitter::emitVecNOrScalar(IRVectorType* vectorType, std::function<void()> emitComponentLogic)
+{
+    if (vectorType)
+    {
+        int N = int(getIntVal(vectorType->getElementCount()));
+        Slang::IRType *elementType = vectorType->getElementType();
+
+        // Special handling required for CUDA target
+        if (isCUDATarget(getTargetReq()))
+        {
+            m_writer->emit("make_");
+
+            switch(elementType->getOp()) 
+            {
+                case kIROp_Int8Type:   m_writer->emit("char"); break;
+                case kIROp_Int16Type:  m_writer->emit("short"); break;
+                case kIROp_IntType:    m_writer->emit("int"); break;
+                case kIROp_Int64Type:  m_writer->emit("longlong"); break;
+                case kIROp_UInt8Type:  m_writer->emit("uchar"); break;
+                case kIROp_UInt16Type: m_writer->emit("ushort"); break;
+                case kIROp_UIntType:   m_writer->emit("uint"); break;
+                case kIROp_UInt64Type: m_writer->emit("ulonglong"); break;
+                default: SLANG_ABORT_COMPILATION("Unhandled type emitting CUDA vector");
+            }
+
+            m_writer->emitRawText(std::to_string(N).c_str());
+        }
+        // Special handling required for Metal target
+        else if (isMetalTarget(getTargetReq()))
```

Comment:
We shouldn't need this, and instead we should make sure the metal backend can emit all vector types correctly.